### PR TITLE
[Merged by Bors] - feat: left adjoint functors preserve sheafification

### PR DESCRIPTION
**Pull request by joelriou**
In this PR, we show that left adjoint functors preserve sheafification. The definition of `Sheaf.composeAndSheafify` was moved to the file `PreservesSheafification.lean` so that this definition may replace the previous `sheafCompose'`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Adjunction/Limits.lean [hunk 1]
```diff
@@ -354,12 +354,12 @@ variable {J C D : Type*} [Category J] [Category C] [Category D]
 noncomputable instance [IsLeftAdjoint F] : PreservesColimitsOfShape J F :=
   (Adjunction.ofIsLeftAdjoint F).leftAdjointPreservesColimits.preservesColimitsOfShape
 
-noncomputable instance [IsLeftAdjoint F] : PreservesColimits F where
+noncomputable instance [IsLeftAdjoint F] : PreservesColimits.{v, u} F where
```
Comment (erdOne) [inline Mathlib/CategoryTheory/Adjunction/Limits.lean:357]:
```suggestion
noncomputable instance [IsLeftAdjoint F] : PreservesColimitsOfSize.{v, u} F where
```
I think?

File: Mathlib/CategoryTheory/Sites/Adjunction.lean [hunk 2]
```diff
@@ -41,16 +40,9 @@ namespace Sheaf
...
-of `X ⋙ G`. -/
-abbrev composeAndSheafify (G : E ⥤ D) : Sheaf J E ⥤ Sheaf J D :=
-  sheafToPresheaf J E ⋙ (whiskeringRight _ _ _).obj G ⋙ presheafToSheaf J D
-set_option linter.uppercaseLean3 false in
-#align category_theory.Sheaf.compose_and_sheafify CategoryTheory.Sheaf.composeAndSheafify
-
 /-- An auxiliary definition to be used in defining `CategoryTheory.Sheaf.adjunction` below. -/
 @[simps]
-def composeEquiv (adj : G ⊣ F) (X : Sheaf J E) (Y : Sheaf J D) :
+def composeEquiv [HasSheafCompose J F](adj : G ⊣ F) (X : Sheaf J E) (Y : Sheaf J D) :
```
Comment (erdOne) [inline Mathlib/CategoryTheory/Sites/Adjunction.lean:45]:
```suggestion
def composeEquiv [HasSheafCompose J F] (adj : G ⊣ F) (X : Sheaf J E) (Y : Sheaf J D) :
```

File: Mathlib/CategoryTheory/Adjunction/Limits.lean (hunk 1 above)
Comment (joelriou) [reply Mathlib/CategoryTheory/Adjunction/Limits.lean:357]: Yes!

**Comment by erdOne** [comment]
Thanks!
maintainer merge

**Pull request closed without merging**